src/uu/cut/src/cut.rs

```diff
@@ -341,43 +341,48 @@ fn cut_files(mut filenames: Vec<String>, mode: Mode) -> i32 {
                 continue;
             }
 
-            exit_code |= match mode {
-                Mode::Bytes(ref ranges, ref opts) => cut_bytes(stdin(), ranges, opts),
-                Mode::Characters(ref ranges, ref opts) => cut_bytes(stdin(), ranges, opts),
-                Mode::Fields(ref ranges, ref opts) => cut_fields(stdin(), ranges, opts),
+            match mode {
+                Mode::Bytes(ref ranges, ref opts) => cut_bytes(stdin(), ranges, opts)?,
+                Mode::Characters(ref ranges, ref opts) => cut_bytes(stdin(), ranges, opts)?,
+                Mode::Fields(ref ranges, ref opts) => cut_fields(stdin(), ranges, opts)?,
             };
 
             stdin_read = true;
         } else {
             let path = Path::new(&filename[..]);
 
+            // Cut operation on potentially multiple files; found errors should not return early,
+            // it should however be displayed with global exit code set for correct exit status.
             if path.is_dir() {
                 show_error!("{}: Is a directory", filename.maybe_quote());
+                set_exit_code(1);
```

Comment:
I was able to remove the majority of the `if` expressions however I think the `is_dir` check may still be required from what I found, the "(os error X)" suffix was being kept for directories after using the suggested`map_err_context`.

I may need a bit of guidance for the proper usage of `UIoError` here as it is a little unclear, but it seems that the `NotADirectory` ErrorKind from `std::io` is currently an unstable feature. This prevented me from manually constructing a relevant `UIoError`. This may be also what is preventing the suffix from being stripped/formatted correctly.

Reply:
If UIoError cannot be mapped here you could also handle the dir check within the match arm. The metadata should be safe to unwrap because the match arm verifies that it is indeed a file:

```rust
match File::open(&path).map_err_context(|| filename.maybe_quote().to_string()) {
    Ok(file) => {
        if file.metadata().unwrap().is_dir() {
            // display appropriate error
            continue;
        }
    // ...
```

Reply:
Ah, I see, that makes sense. We'll just wait until `NotADirectory` is no longer unstable then. I think it's best to leave the manual check for now. Thank you for looking into this!